rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +481,337 @@ def accepted_type(self) -> type:
+                and field.typeval.size != field.length.value
+            ):
+                field.first = UNDEFINED
+                field.length = UNDEFINED
+                field.typeval.clear()
+                break
+
+            self._last_field = nxt
+            nxt = self._next_field(nxt)
+
+    def get(self, fld: str) -> Any:
+        if fld not in self.valid_fields:
+            raise ValueError(f"field {fld} not valid")
+        return self._fields[fld].typeval.value
+
+    @property
+    def bitstring(self) -> Bitstring:
+        bits = ""
+        field = self._next_field(INITIAL.name)
+        while field and not field == FINAL.name:
+            field_val = self._fields[field]
+            if (
+                not field_val.set
+                or not isinstance(field_val.first, Number)
+                or not field_val.first.value <= len(bits)
+            ):
+                break
+            bits = bits[: field_val.first.value] + str(self._fields[field].typeval.bitstring)
+            field = self._next_field(field)
+        if len(bits) % 8:
+            raise ValueError(f"message length must be dividable by 8 ({len(bits)})")
+
+        return Bitstring(bits)
+
+    @property
+    def value(self) -> Any:
+        raise NotImplementedError
+
+    @property
+    def bytestring(self) -> bytes:
+        bits = str(self.bitstring)
+        return b"".join(
+            [int(bits[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(bits), 8)]
+        )
+
+    @property
+    def fields(self) -> List[str]:
+        return [f.name for f in self._model.fields]
+
+    @property
+    def accessible_fields(self) -> List[str]:
+
+        nxt = self._next_field(INITIAL.name)
+        fields: List[str] = []
+        while nxt and nxt != FINAL.name:
+
+            if (
+                self.__simplified(self._model.field_condition(Field(nxt))) != TRUE
+                or not self._has_first(nxt)
+                or (
+                    not self._has_length(nxt)
+                    if not isinstance(self._fields[nxt].typeval, OpaqueValue)
+                    else self._is_valid_opaque_node(nxt)
+                )
+            ):
+                break
+
+            fields.append(nxt)
+            nxt = self._next_field(nxt)
+        return fields
+
+    def _is_valid_opaque_node(self, nxt: str) -> bool:
+
+        if self._get_length_unchecked(nxt) == UNDEFINED:
+            return True
+
+        for edge in self._model.incoming(Field(nxt)):
+            if self.__simplified(edge.condition) == TRUE:
+                valid_edge = edge
+                break
+        else:
+            return False
+
+        for ve in valid_edge.length.variables():
+            assert isinstance(ve.name, str)
+            if ve.name in self._fields and not self._fields[ve.name].set:
+                return True
+
+            if isinstance(ve, Last) and ve.name == "Message":
+                return False
+
+        return False
```

Comment:
1. `nxt` is a odd parameter name.
2. Why is a opaque node valid, if its length is undefined?

Reply:
The method returns False if the field **is** valid. Since this is a bit counterintuitive i changed True to False and vice versa and renamed it again as _is_valid_opaque_field because node is not used anywhere else.

Reply:
Fixed by 301b231